www/js/lib/filecache.js

```diff
@@ -36,21 +36,36 @@ define(['q'], function (Q) {
      */
     const BLOCK_SIZE = 4096;
 
+    /**
+     * A Cache Entry
+     * @typedef CacheEntry
+     * @property {String} id The cache key (stored also in the entry)
+     * @property {CacheEntry} prev The previous linked cache entry
+     * @property {CacheEntry} next The next linked cache entry
+     * @property {Uint8Array} value The cached data
+     */
+
+    /**
+     * A Block Cache employing a Least Recently Used caching strategy
+     * @typedef BlockCache
+     * @property {Integer} _limit The maximum number of entries in the cache 
+     * @property {Map} _entries A map to store the cache keys and data
+     * @property {CacheEntry} _first The most recent entry in the cache
+     * @property {CacheEntry} _last The least recedntly used entry in the cache
+     */
+    
     /**
      * Creates a new cache with max size limit of MAX_CACHE_SIZE blocks
      */
     function LRUCache() {
         console.log('Creating cache of size ' + MAX_CACHE_SIZE + ' * ' + BLOCK_SIZE + ' bytes');
         this._limit = MAX_CACHE_SIZE;
-        // linked list of entries
-        this._first = null;
-        this._last = null;
     }
 
     /**
      * Tries to retrieve an element by its id. If it is not present in the cache, returns undefined; if it is present,
      * then the value is returned and the entry is moved to the top of the cache
-     * @param {String} key The block cache entry key (file.id + ':' + byte offset)
+     * @param {String} key The block cache entry key (byte offset + '' + file.id)
```

Comment:
The file id allows for a more simple separator : that's a good idea!
But I think the separator should be kept : even if it's unlikely, there could be a mix between 1+''+11 and 11+''+1 (for example). I think it's better to have a 100% safe key (with a separator like `:`) than to save only 4kB (without a separator)

Reply:
He he, you never miss anything, @mossroy! I agree there is a theoretical ambiguity in the key format. In practice, the ambiguity should never occur, because the smallest ambiguous number is (I think) 110, which could be 11:0 or 1:10. However, we void `cache._entries` (the Map) on each ZIM change, so the only entries from a previous ZIM that can remain in the cache would be assets still resolving from the previous ZIM load (which woudl be numbered 9 in the case of 1:10).  Nevertheless, it is a small cost to remove any possibility of ambiguity, as you say, and I have restored the separator. Thank you for your eagle eye!